pkg/jwt/private_keys.go

```diff
@@ -0,0 +1,47 @@
+package jwt
+
+import (
+	"crypto"
+
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+// PrivateKeyGetter stores JWT private keys.
+// Note: the reason we use crypto.Signer here is because crypto.PrivateKey
+// is empty interface that does not implement Public() method.
+type PrivateKeyGetter interface {
```

Comment:
Somehow the names do not match, if I have a Getter I wouldn't expect any Update functionality exposed in the interface. This would be more of a ReadWriter kind of interface, if we want to comply with examples for io packages.

And additionally: I think we should split those interfaces, one interface which only allows Read / Get, and the other which allows Write / Set.
The public interface for the validator and signer should IMO only read from the key set, since it doesn't require write operations. The write is only required in the case of central initializing a JWT key store which also can update the underlying key. That feels much more in line with the overall functionality as well as the already defined interfaces.

Reply:
Good thinking! IMHO words `Read` and `Write` are strongly associated files or other external entities, so I kept `Getter/Setter` as the names for interfaces.